crates/bevy_ecs/src/event.rs

```diff
@@ -284,10 +284,19 @@ impl<'w, 's, T: Resource> EventReader<'w, 's, T> {
         self.events.event_reader_len(self.last_event_count.0)
     }
 
-    /// Determines if are any events available to be read without consuming any.
+    /// Determines if no events are available to be read without consuming any.
+    /// If you need to consume the iterator you can use [`EventReader::any`]
+    /// WARNING: `events.is_empty()` is not the same as doing `!events.any()`
     pub fn is_empty(&self) -> bool {
         self.len() == 0
     }
+
+    /// Determines if there are any events available to be read and consumes all events.
+    /// If you don't need to consume the iterator you can use [`EventReader::is_empty`]
+    /// WARNING: `events.any()` is not the same as doing `!events.is_empty()`
+    pub fn any(&mut self) -> bool {
```

Comment:
Would taking self by value be possible? That avoids accidentally using if after calling `.any()`.

Reply:
I tested it and that would work, but I think I will go with the approach suggested by @devil-ira that only uses a separate clear method.